tests/integration/test_auth.py

```diff
@@ -0,0 +1,141 @@
+import base64
+import http.client
+import os
+import platform
+import shutil
+import signal
+import subprocess
+import tempfile
+import time
+import unittest
+
+
+class TestAuth(unittest.TestCase):
+    def setUp(self):
+        self.payload = (
+            '''{
+                "data": { "_arg1": [1, 2] },
+                "script": "return [x * 2 for x in _arg1]"
+            }''')
+
+        prefix = 'TabPyIntegTest'
+        self.tmp_dir = tempfile.mkdtemp(prefix=prefix)
+
+        # create temporary state.ini
+        self.state_file = open(os.path.join(self.tmp_dir, 'state.ini'), 'w+')

```

Comment:
rather than creating a temporary state.ini and config for each test run, should we just keep local copies in the test folder?

Reply:
One reason I didn't do it is we can move this code in some integration test base class to be reused. This is why I want to keep it as generic as possible right now - we will need new state.ini for models tests.